src/core/shape_inference/include/detection_output_shape_inference.hpp

```diff
@@ -168,6 +168,20 @@ void shape_infer_base(const DetectionOutputBase* op,
                                   ".");
             num_prior_boxes = box_logits_pshape_2nd_dim / (num_loc_classes * 4);
         }
+
+        if (num_prior_boxes > 0 && num_loc_classes > 0) {
+            NODE_VALIDATION_CHECK(op,
```

Comment:
Please refactor this validation to reduce number arguments. generating this kind of validation has negative impact on binary size.
e.g:
the `but must be compatible with (num_prior_boxes * num_loc_classes * 4) = (",`, can be removed  as check `box_logits_pshape[1].compatible(num_prior_boxes * num_loc_classes * 4)` will converted to string.

Instead interupt string with values try make "explanation", current value, expect value etc.

` box_logits_pshape[1],` printing it could be replaced by new macro `NODE_SHAPE_INFER_CHECK`

Reply:
Ok, I wanted this error message to be as much informative as possible, but I have shortened it.